mongodb-community-operator/controllers/validation/validation.go

```diff
@@ -83,16 +83,18 @@ func validateUsers(mdb mdbv1.MongoDBCommunity) error {
 			connectionStringSecretNameMap[connectionStringSecretName] = user
 		}
 
-		// Ensure no collisions in the secret holding scram credentials
-		scramSecretName := user.ScramCredentialsSecretName
-		if previousUser, exists := scramSecretNameMap[scramSecretName]; exists {
-			scramSecretNameCollisions = append(scramSecretNameCollisions,
-				fmt.Sprintf(`[scram secret name: "%s" for user: "%s" and user: "%s"]`,
-					scramSecretName,
-					previousUser.Username,
-					user.Username))
-		} else {
-			scramSecretNameMap[scramSecretName] = user
+		if user.Database != constants.ExternalDB {
+			// Ensure no collisions in the secret holding scram credentials
+			scramSecretName := user.ScramCredentialsSecretName
```

Comment:
q: why not check if the `user.ScramCredentialsSecretName` is not empty? I think this is directly related to the cause of the issue and the `user.Database != constants.ExternalDB` is not so self explanatory